frontend/src/services/connectionTypesService.ts

```diff
@@ -0,0 +1,78 @@
+import axios from '~/utilities/axios';
+import { ResponseStatus } from '~/types';
+import {
+  ConnectionTypeConfigMap,
+  ConnectionTypeConfigMapObj,
+} from '~/concepts/connectionTypes/types';
+import {
+  toConnectionTypeConfigMap,
+  toConnectionTypeConfigMapObj,
+} from '~/concepts/connectionTypes/utils';
+
+export const fetchConnectionTypes = (): Promise<ConnectionTypeConfigMapObj[]> => {
+  const url = `/api/connection-types`;
+  return axios
+    .get(url)
+    .then((response) =>
+      response.data.map((cm: ConnectionTypeConfigMap) => toConnectionTypeConfigMapObj(cm)),
+    )
+    .catch((e) => {
+      throw new Error(e.response.data.message);
+    });
+};
+
+export const fetchConnectionType = (name: string): Promise<ConnectionTypeConfigMapObj> => {
+  const url = `/api/connection-types/${name}`;
+  return axios
+    .get(url)
+    .then((response) => toConnectionTypeConfigMapObj(response.data))
+    .catch((e) => {
+      throw new Error(e.response.data.message);
+    });
+};
+
+export const createConnectionType = (
+  connectionType: ConnectionTypeConfigMapObj,
+): Promise<ResponseStatus> => {
+  const url = `/api/connection-types`;
+  return axios
+    .post(url, toConnectionTypeConfigMap(connectionType))
+    .then((response) => response.data)
+    .catch((e) => {
+      throw new Error(e.response.data.message);
+    });
+};
+
+export const updateConnectionType = (
+  connectionType: ConnectionTypeConfigMapObj,
+): Promise<ResponseStatus> => {
+  const url = `/api/connection-types/${connectionType.metadata.name}`;
+  return axios
+    .put(url, toConnectionTypeConfigMap(connectionType))
+    .then((response) => response.data)
+    .catch((e) => {
+      throw new Error(e.response.data.message);
+    });
+};
+
+export const patchConnectionType = (
+  connectionType: ConnectionTypeConfigMapObj,
+): Promise<ResponseStatus> => {
```

Comment:
Let's omit this generic patch function and add a specific one that constructs a patch just for setting the `enabled` annotation:
```
             {
                op: 'replace',
                path: '/metadata/annotations/opendatahub.io~1enabled',
                value: ...,
              }
```